cdap-metadata-spi/src/main/java/io/cdap/cdap/spi/metadata/MetadataConsumer.java

```diff
@@ -24,9 +24,15 @@
 @Beta
 public interface MetadataConsumer {
 
+  /**
+   * @return the name of the metadata consumer.
+   */
+  String getName();
+
   /**
    * Consumes lineage info for a program run.
    */
-  void consumeLineage(ProgramRun programRun, LineageInfo lineageInfo);
+  void consumeLineage(MetadataConsumerContext context, ProgramRun programRun, LineageInfo lineageInfo)
```

Comment:
minor nit: Usually for javadoc of the method we add description of all parameters and exception thrown (about when and why)